### PR TITLE
Add schema-based rules for platform data of package index

### PR DESCRIPTION
**Pull request by per1234**
These are the JSON schema-based rules to check the platform level data.

### CODE REVIEW

File: internal/rule/ruleconfiguration/ruleconfiguration.go [hunk 1]
```diff
@@ -3305,37 +3305,917 @@ var configurations = []Type{
...
+		RuleFunction:     rulefunction.PackageIndexPackagesPlatformsMissing,
+	},
+	{
+		ProjectType:      projecttype.PackageIndex,
+		SuperprojectType: projecttype.All,
+		Category:         "data",
+		Subcategory:      "platform",
+		ID:               "IL002",
+		Brief:            "incorrect packages[].platforms type",
+		Description:      "Must be a string.",
```
Comment (silvanocerza) [inline internal/rule/ruleconfiguration/ruleconfiguration.go:3331]: Is this correct? Shouldn't `packages[].platforms` be a json array? :thinking:

File: internal/rule/ruleconfiguration/ruleconfiguration.go [hunk 2]
```diff
@@ -3305,37 +3305,917 @@ var configurations = []Type{
...
+		RuleFunction:     rulefunction.PackageIndexPackagesPlatformsBoardsMissing,
+	},
+	{
+		ProjectType:      projecttype.PackageIndex,
+		SuperprojectType: projecttype.All,
+		Category:         "data",
+		Subcategory:      "platform",
+		ID:               "IL040",
+		Brief:            "incorrect packages[].platforms[].boards type",
+		Description:      "Must be an object.",
```
Comment (silvanocerza) [inline internal/rule/ruleconfiguration/ruleconfiguration.go:3939]: Shouldn't it be an array?

File: internal/rule/ruleconfiguration/ruleconfiguration.go [hunk 3]
```diff
@@ -3305,37 +3305,917 @@ var configurations = []Type{
...
+		RuleFunction:     rulefunction.PackageIndexPackagesPlatformsToolsDependenciesMissing,
+	},
+	{
+		ProjectType:      projecttype.PackageIndex,
+		SuperprojectType: projecttype.All,
+		Category:         "data",
+		Subcategory:      "platform",
+		ID:               "IL046",
+		Brief:            "incorrect packages[].platforms[].toolsDependencies type",
+		Description:      "Must be an object.",
```
Comment (silvanocerza) [inline internal/rule/ruleconfiguration/ruleconfiguration.go:4035]: This one too.

File: internal/rule/ruleconfiguration/ruleconfiguration.go (hunk 1 above)
Comment (per1234) [reply internal/rule/ruleconfiguration/ruleconfiguration.go]: Thanks Silvano! All three fixed here:
https://github.com/arduino/arduino-lint/compare/6af76b178f491b9239c734d43bb09613c81f9e84..c1371a1e713cc0918c4ecb67da6d4fce0221fcdc

**Review by silvanocerza — APPROVED** [review]

**Pull request merged**